### PR TITLE
feat(chore): Add latest version of AA circuits

### PR DESCRIPTION
**Pull request by Meyanis95**
This adds the latest version of the AA protocol circuits. 

zkey file: 1.05GB
Number of constraints: 1,767,153

- A test has been added in `mopro-core`
- Added some hardcoded variables to generate dylib (might remove them if needed) 
- This has not been tested on mobile

### CODE REVIEW

File: mopro-core/build.rs [hunk 1]
```diff
@@ -79,8 +79,10 @@ fn build_dylib(wasm_path: String, dylib_name: String) -> Result<()> {
 
 fn main() -> Result<()> {
```
Comment (oskarth) [inline mopro-core/build.rs:80]: This is very messy... I'm working on making this easier to work with. Will update once fixed, then we can make this easier to merge

File: mopro-ios/MoproKit/Example/MoproKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -261,6 +264,7 @@
 					};
 					607FACE41AFB9204008FA782 = {
 						CreatedOnToolsVersion = 6.3.1;
+						DevelopmentTeam = 7BMWCJP4MA;
```
[the remark this answers is not in the document]
Comment (Meyanis95) [reply mopro-ios/MoproKit/Example/MoproKit.xcodeproj/project.pbxproj:267]: Yeah, I think that's Xcode that modifies that, I'll remove it and the others.

File: mopro-core/build.rs (hunk 1 above)
Comment (Meyanis95) [reply mopro-core/build.rs:80]: Ok let me know, how I should resolve once done

Comment (oskarth) [reply mopro-core/build.rs:80]: Fixed: https://github.com/oskarth/mopro/pull/67/files#diff-10095d80195138814b66f1d6f5b769b0763cd537d9ef563a28c6d75444a151d5

See https://github.com/oskarth/mopro/pull/67/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R34 for what to do

I suggest adding an `anon-aadhaar-config.toml` to this PR

**Comment by oskarth** [comment]
Tested in Xcode Simulator and getting: `dylib feature not enabled!` error

This is due to it calling `initialize_mopro_dylib` without dylib feature.

I think for simplicity we can start without the dylib, comment out that dylib init block in iOS and use vanilla wasm (like other examples)? IIRC the main impact is that you can't disconnect cable, but perf should be basically the same. WDYT?

**Comment by Meyanis95** [comment]
- Added config file `anon-aadhaar-config.toml`
- Removed dylib from the build config
- Rebased from main
- Updated Swift view with the right circuit params
- Added `test_setup_prove_anon_aadhaar()` test in `mopro-core`

I was able to run aa prover in the simulator with the following results:

Screenshot 2024-02-02 at 11 40 33

**Review by oskarth — APPROVED** [review]
Awesome! Let me try

**Comment by oskarth** [comment]
Ah ti seems like CI is failing

```
       }
 
-        let file_data = std::fs::read_to_string("./examples/circom/anonAadhaar/input.json").expect("Unable to read file");
-        let data: InputData = serde_json::from_str(&file_data).expect("JSON was not well-formatted");
+        let file_data = std::fs::read_to_string("./examples/circom/anonAadhaar/input.json")
+            .expect("Unable to read file");
+        let data: InputData =
+            serde_json::from_str(&file_data).expect("JSON was not well-formatted");
 
         let mut inputs: CircuitInputs = HashMap::new();
         inputs.insert(
Diff in /home/runner/work/mopro/mopro/mopro-core/src/middleware/circom/mod.rs at line 742:
             pub_key: Vec<String>,
         }
 
-        let file_data = std::fs::read_to_string("./examples/circom/anonAadhaar/input.json").expect("Unable to read file");
-        let data: InputData = serde_json::from_str(&file_data).expect("JSON was not well-formatted");
+        let file_data = std::fs::read_to_string("./examples/circom/anonAadhaar/input.json")
+            .expect("Unable to read file");
+        let data: InputData =
+            serde_json::from_str(&file_data).expect("JSON was not well-formatted");
 
         let mut inputs: CircuitInputs = HashMap::new();
         inputs.insert(
Error: Process completed with exit code 1.
```

and

```
Command line invocation:
    /Applications/Xcode_14.2.app/Contents/Developer/usr/bin/xcodebuild test -scheme MoproKit-Example -workspace MoproKit.xcworkspace -destination "platform=iOS Simulator,OS=16.2,name=iPhone 14 Pro"

User defaults from command line:
    IDEPackageSupportUseBuiltinSCM = YES

Prepare packages
2024-02-02 11:07:08.766 xcodebuild[10001:60376] Writing error result bundle to /var/folders/qv/pdh5wsgn0lq3dp77zj602b5c0000gn/T/ResultBundle_2024-02-02_11-07-0008.xcresult

xcodebuild: error: Failed to build workspace MoproKit with scheme MoproKit-Example.
	Reason: Cannot test target “MoproKit_Tests” on “iPhone 14 Pro”: iPhone 14 Pro’s iOS Simulator 16.2 doesn’t match MoproKit_Tests’s iOS Simulator 17.2 deployment target.
Error: Process completed with exit code 70.
```

(also Android but think this is unrelated to PR)

**Comment by oskarth** [comment]
(Also on simulator):

```
Initializing library with arkzkey
Time to deserialize proving key: 82.991869583s
Time to deserialize matrices: 1.248941541s
Initializing arkzkey took: 84.24s
Generating proof 2
Witness generation took: 10.47s
Loading arkzkey took: 166.00ns
proof generation took: 18.99s
Verification time 2: 4.077041ms
MoproKit_Example/AnonAadhaarViewControllerNew.swift:1766: Assertion failed: Proof verification should succeed
```

Proof verification doesn't seem to work, is this to be expected with dummy data or so?

**Comment by Meyanis95** [comment]
Regarding the first issue with JSON loading, it's weird because these tests are marked as being ignored by the CI, so this piece of code shouldn't be triggered by the CI workflow.

File: anon-aadhaar-config.toml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# anon-aadhaar-config.toml
+
+[build]
+# For iOS device_type can be x86_64, simulator, device
+# For Android device_type can be x86_64, arm, arm64
+device_type = "simulator" # Options: x86_64, simulator, device, arm, arm64
```
Comment (vivianjeng) [inline anon-aadhaar-config.toml:6]: please use `x86_64` for ci tests

File: scripts/update_bindings.sh [hunk 4]
```diff
@@ -153,4 +153,4 @@ if [[ "$USE_DYLIB" == true ]]; then
     codesign -f -s "${APPLE_SIGNING_IDENTITY}" "${MOPROKIT_DIR}/Libs/${DYLIB_NAME}"
 fi
 
-print_action "Done! Please re-build your project in Xcode."
\ No newline at end of file
+print_action "Done! Please re-build your project in Xcode."
```
Comment (vivianjeng) [inline scripts/update_bindings.sh:156]: should be a new line here?

**Review by vivianjeng — COMMENTED** [review]
please run `cargo fmt --all` in `mopro-core`

File: scripts/update_bindings.sh (hunk 4 above)
Comment (Meyanis95) [reply scripts/update_bindings.sh]: Not sure.
But I removed it

File: mopro-ios/MoproKit/Example/MoproKit.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -624,8 +630,9 @@
 					"$(inherited)",
 				);
 				INFOPLIST_FILE = Tests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 17.2;
```
Comment (vivianjeng) [inline mopro-ios/MoproKit/Example/MoproKit.xcodeproj/project.pbxproj:633]: try not to specify this?

File: mopro-ios/MoproKit/Example/MoproKit.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -604,8 +609,9 @@
...
```
Comment (vivianjeng) [inline mopro-ios/MoproKit/Example/MoproKit.xcodeproj/project.pbxproj:612]: also here 🙏🏻

**Pull request merged**